### PR TITLE
Fixed user reset events

### PR DESCRIPTION
**Pull request by gonzalovilaseca**
More info about this PR here: https://github.com/Sylius/Sylius/issues/3167
I'm happy to change the events description if anyone has better ideas.

### CODE REVIEW

**Comment by gonzalovilaseca** [comment]
Travis is throwing an error on search test, I will have a look this evening.

File: src/Sylius/Bundle/UserBundle/UserEvents.php [hunk 1]
```diff
@@ -31,19 +31,33 @@
...
      */
-    const PASSWORD_CHANGE_SUCCESS = 'sylius.user.password_change.success';
+    const POST_PASSWORD_RESET = 'sylius.user.password_reset.post';
+    /**
+     * The PRE_PASSWORD_CHANGE event occurs before the change form is submitted successfully.
+     *
+     * This event allows you to set the response instead of using the default one.
+     * The event listener method receives a Symfony\Component\EventDispatcher\GenericEvent instance.
+     */
+    const PRE_PASSWORD_CHANGE = 'sylius.user.password_change.pre';
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/UserBundle/UserEvents.php]: Missing blank line after this.

File: src/Sylius/Bundle/UserBundle/UserEvents.php [hunk 2]
```diff
@@ -31,19 +31,33 @@
...
+    /**
+     * The POST_PASSWORD_CHANGE event occurs after the change form is submitted successfully.
+     *
+     * This event allows you to set the response instead of using the default one.
+     * The event listener method receives a Symfony\Component\EventDispatcher\GenericEvent instance.
+     */
+    const POST_PASSWORD_CHANGE = 'sylius.user.password_change.post';
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/UserBundle/UserEvents.php]: Missing blank line as well.

File: src/Sylius/Bundle/UserBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -124,8 +124,8 @@
         </service>
         <service id="sylius.listener.password_updater" class="%sylius.listener.password_updater.class%">
             <argument type="service" id="sylius.user.password_updater" />
-            <tag name="kernel.event_listener" event="sylius.user.password_reset.success" method="genericEventUpdater" />
-            <tag name="kernel.event_listener" event="sylius.user.password_change.success" method="genericEventUpdater" />
+            <tag name="kernel.event_listener" event="sylius.user.password_reset.pre" method="genericEventUpdater" />
+            <tag name="kernel.event_listener" event="sylius.user.password_change.pre" method="genericEventUpdater" />
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/UserBundle/Resources/config/services.xml]: I'd rename these events to: ``sylius.user.pre/post_password_change``. This is the convention we have for other events.

**Comment by pjedrzejewski** [comment]
@gonzalovilaseca Thanks man! I restarted the build - we need to track down the random bug with search... Please have a look at my comments.

**Comment by gonzalovilaseca** [comment]
@pjedrzejewski NP! I'll update PR 2mro with sugested changes.

**Comment by gonzalovilaseca** [comment]
PR updated, once it's ready to merge I'll rebase it.

**Comment by michalmarcinkowski** [comment]
Great! Thank you Gonzalo! :+1:

**Pull request merged**